app/src/main/java/com/example/plantsapp/presentation/ui/worker/ReschedulingWorker.kt

```diff
@@ -24,10 +24,13 @@ class ReschedulingWorker @AssistedInject constructor(
     override suspend fun doWork(): Result {
 
         plantsRepository.fetchPlants()
-            .map { plant -> getTasksForPlantAndDateUseCase(plant, Date().minusDays(1)) }
-            .flatten()
-            .filter { (_, isCompleted) -> !isCompleted }
-            .forEach { (task, _) -> tasksRepository.updateTask(task, Date()) }
+            .forEach { plant ->
+                getTasksForPlantAndDateUseCase(plant, Date().minusDays(1))
+                    .filter { (_, isCompleted) -> !isCompleted }
+                    .forEach { (task, _) ->
+                        tasksRepository.updateTask(plant, task, Date())
+                    }
+            }
```

Comment:
1. How about cache Date outside of scope? Because for each task you will have different date right now
2. maybe return map + flatten?

Reply:
> 

But we need Plant for updateTask() now